test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,696 @@
+
+void testViewCreationChar(string viewName, string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testPrimaryKeysChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, string schemaName, string pkTableName, vector<string> primaryKeyColumns, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testUniqueConstraintChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> uniqueConstraintColumns, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testComparisonOperators(string tableName, vector<pair<string, string>> tableColumns, string col1Name, string col2Name, vector<string> col1Data, vector<string> col2Data, vector<string> operationsQuery, vector<vector<char>> expectedResults);
+
+void testComparisonFunctionsChar(string tableName, vector<pair<string, string>> tableColumns, vector<string> insertedData, vector<string> operationsQuery, vector<string> expectedResults);
+
+template <typename T>
+void testInsertionSuccessNumeric(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, int type, T data, int bufferLen, vector<string> insertedValues, vector<T> expectedInsertedValues, vector<long> expectedLen) {
+  
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  RETCODE rcode;
+  SQLLEN affectedRows;
+  int pk;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> bind_columns = {
+    {1, SQL_C_LONG, &pk, 0, &pk_len},
+    {2, type, &data, bufferLen, &data_len}
+  };
+
+  string insertString{};
+  string comma{};
+  for (int i = 0; i < insertedValues.size(); i++) {
+      insertString += comma + "(" + std::to_string(i) + "," + insertedValues[i] + ")";
+      comma = ",";
+  }
+
+  // Create table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(tableName, tableColumns));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(tableName, insertString));
+  
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affectedRows);
+  EXPECT_EQ(rcode, SQL_SUCCESS);
+  EXPECT_EQ(affectedRows, insertedValues.size());
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(tableName, {"*"}, vector<string> {orderByColumnName}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < insertedValues.size(); i++) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(pk_len, 4);
```

Comment:
What is `4` suppose to represent? Can we extract this to a variable

Reply:
Similarly to other places too.

Reply:
4 is supposed to represent the length of the primary key, which is an int4 type. Done, extracted to variable `INT_BYTES_EXPECTED` in `psqlodbc_tests_common.h`.